### DISCUSSION TITLE
TypeError and  "Could not find plugin" when using flat config

### DISCUSSION THREAD

**Discussion by tomdavidson** [eslint/eslint#16456] [Q&A] [answered]
Hello, I have moved the flat config and ESLint 8.26.0 but the functional plugin can not be resolved.  I used pnpm and eslint-plugin-functional prior to flat config just fine.

```shell
TypeError: Key "rules": Key "functional/prefer-tacit": Could not find plugin "functional".
    at throwRuleNotFoundError (./node_modules/.pnpm/eslint@8.26.0/node_modules/eslint/lib/config/rule-validator.js:65:11)
```

But the module is present:
```shell
$ ls node_modules/eslint-plugin-functional
CHANGELOG.md  lib  LICENSE  node_modules  package.json  README.md
```

Here is a gist with the [eslint.config.js](https://gist.github.com/tomdavidson/e7ac825e66201259713c1b47eb997ed3) (I get these same regarless of type module or commonjs)

FYI I had asked for help on this in discord too.

**Comment by mdjermanovic** [comment, marked as answer]
This happens because [`func.configs.stylistic.rules`](https://gist.github.com/tomdavidson/e7ac825e66201259713c1b47eb997ed3#file-eslint-config-js-L302) enables rule `"functional/prefer-tacit"` but there's no `"functional"` key in your [`plugins`](https://gist.github.com/tomdavidson/e7ac825e66201259713c1b47eb997ed3#file-eslint-config-js-L286-L290) (you're using `"func"` key instead).

**Reply by tomdavidson** [reply]
bloody heck, that's so obvious! (now that you have pointed it out)

Thank you.
